code-execution-engines/langchain-code-execution-engine-benchmark/src/test/java/dev/langchain/code/benchmark/JMHBenchmarkJavaScriptExecution.java

```diff
@@ -0,0 +1,70 @@
+package dev.langchain.code.benchmark;
+
+
+import dev.langchain4j.code.CodeExecutionEngine;
+import dev.langchain4j.code.graalvm.GraalVmJavaScriptExecutionEngine;
+import dev.langchain4j.code.v8.V8JavaScriptExecutionEngine;
+import org.openjdk.jmh.annotations.*;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.CommandLineOptionException;
+import org.openjdk.jmh.runner.options.CommandLineOptions;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+
+import java.util.concurrent.TimeUnit;
+
+@State(Scope.Benchmark)
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@Fork(1)
+@Warmup(iterations = 1)
+@Measurement(iterations = 2)
+public class JMHBenchmarkJavaScriptExecution {
+
+
+    private String code;
+    private CodeExecutionEngine v8Engine;
+    private CodeExecutionEngine graalEngine;
+
+
+    @Setup(Level.Trial)
+    public void setUp(){
+        code = """
+            function fibonacci(n) {
+                    if (n <= 1) return n;
+                    return fibonacci(n - 1) + fibonacci(n - 2);
+                }
+                                
+            fibonacci(15)
+            """;
+
+        graalEngine = new GraalVmJavaScriptExecutionEngine();
+        v8Engine = new V8JavaScriptExecutionEngine();
+    }
+
+
+    @Benchmark
+    public void benchmarkGraal(){
+        graalEngine.execute(code);
+    }
+
+
+    @Benchmark
+    public void benchmarkV8(){
+        v8Engine.execute(code);
+    }
+
+
+    public static void main(String[] args) throws RunnerException, CommandLineOptionException {
```

Comment:
Is this module really needed?

Reply:
I think for benchmarks code engines in the future it's nice to have

anyone think different ? I can emit it for now

Reply:
This can be useful for code engine implementors, but there are many different reasons to pick an engine over another one and not just performance.
I don't think we should ship this as a module.